pennylane_lightning_gpu/src/simulator/StateVectorCudaManaged.hpp

```diff
@@ -806,7 +810,6 @@ class StateVectorCudaManaged
         PL_CUDA_IS_SUCCESS(cudaMalloc(&dBuffer, bufferSize))
```

Comment:
Can this also be moved to a DataBuffer type?

Reply:
Tried but it didn't work when I set the type of void. Need to figure it later.

Reply:
The issue here is that it will outright fail on a multiGPU device workload. The buffer handles automatic movement between GPUs if required, and will help pinning the device index for such operations.

Reply:
If we need to update the buffer to support the required types, happy to do so.

Reply:
Yes, updating the buffer implementation now. It works for FP32 but fails for FP64 and figuring why now.

Reply:
Works now after set alpha & beta as CFP_t.